client/src/reducers/controls.js

```diff
@@ -95,61 +94,57 @@ const Controls = (
         state,
         world
       );
-      const crossfilter = Crossfilter(world.obsAnnotations);
-      const dimensionMap = World.createObsDimensionMap(crossfilter, world);
+      const crossfilter = World.createObsDimensions(
+        new Crossfilter(world.obsAnnotations),
+        world
+      );
       WorldUtil.clearCaches();
 
       return {
         ...state,
         loading: false,
         error: null,
         universe,
-        fullUniverseCache: { world, crossfilter, dimensionMap },
+        fullUniverseCache: { world, crossfilter },
         world,
         categoricalSelectionState,
         crossfilter,
-        dimensionMap,
         colorMode,
         colorAccessor: null,
         colors,
         resettingInterface: false
       };
     }
     case "reset World to eq Universe": {
+      /*
+      1. Reset world & crossfilter, using previously created objects which were
+         stashed in `fullUniverseCache`
+      2. Add crossfilter dimension for all userDefined and diffexp genes/varData,
+         as they are not part of the cached crossfilter.
+      3. Compute categorical selection summary
+      4. Reset all WorldUtil caches
+      5. Reset color-by
+      */
       const { userDefinedGenes, diffexpGenes, fullUniverseCache } = state;
-      const { world, crossfilter } = fullUniverseCache;
-      // reset all crossfilter dimensions
-      _.forEach(fullUniverseCache.dimensionMap, dim => dim.filterAll());
+      const { world } = fullUniverseCache;
+      const crossfilter = ControlsHelpers.createGeneDimensions(
+        userDefinedGenes,
+        diffexpGenes,
+        world,
+        fullUniverseCache.crossfilter
+      );
       const colorMode = null;
       const colors = createColors(world, colorMode);
       const categoricalSelectionState = ControlsHelpers.createCategoricalSelectionState(
         state,
         world
       );
-
-      /* free dimensions not in cache (otherwise they leak) */
-      _.forEach(state.dimensionMap, (dim, dimName) => {
-        if (!fullUniverseCache.dimensionMap[dimName]) {
-          dim.dispose();
-        }
-      });
-      const dimensionMap = {
```

Comment:
Confirming: dimensionMap as a concept is no longer around?

Reply:
correct.  The equivalent is now part of the crossfilter object itself, and you reference dimensions by name.

Eg,  `crossfilter.addDimension("foo", ...)`